libkpod/pause.go

```diff
@@ -0,0 +1,44 @@
+package libkpod
+
+import (
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/pkg/errors"
+)
+
+// ContainerPause pauses a running container.
+func (c *ContainerServer) ContainerPause(container string) (string, error) {
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to find container %s", container)
+	}
+
+	cStatus := c.runtime.ContainerStatus(ctr)
+	if cStatus.Status != oci.ContainerStatePaused {
+		if err := c.runtime.PauseContainer(ctr); err != nil {
+			return "", errors.Wrapf(err, "failed to pause container %s", ctr.ID())
+		}
+	}
+
+	c.ContainerStateToDisk(ctr)
+
+	return ctr.ID(), nil
+}
+
+// ContainerUnpause unpauses a running container with a grace period (i.e., timeout).
+func (c *ContainerServer) ContainerUnpause(container string) (string, error) {
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to find container %s", container)
+	}
+
+	cStatus := c.runtime.ContainerStatus(ctr)
+	if cStatus.Status == oci.ContainerStatePaused {
```

Comment:
I think it's worth returning an error if we're not paused here

Reply:
I've added the error but was kind of on the fence about not doing so as the ending state (unpaused) would be the same, i.e. the container would be unpaused as we requested.  But maybe it will prove useful to have.  Similar thought on pause, fwiw.

Reply:
I'm not strongly convinced it's the right thing to do, but Docker returns an error for this case, so I think it's probably worth including.